firebase-storage/ktx/src/main/kotlin/com/google/firebase/storage/ktx/Storage.kt

```diff
@@ -133,6 +140,93 @@ operator fun ListResult.component2(): List<StorageReference> = prefixes
  */
 operator fun ListResult.component3(): String? = pageToken
 
+/**
+ * Starts listening to this task's stream download progress and emits its values via a [Flow].
+ *
+ * - When the returned flow starts being collected, it attaches the following listeners:
+ * [OnPausedListener], [OnProgressListener].
+ * - When the flow completes the listeners will be removed.
+ */
+val StreamDownloadTask.progress: Flow<StorageProgress<StreamDownloadTask.TaskSnapshot>>
```

Comment:
Looks like these properties share a lot in common, have you considered implementing this property for `StorageTask` that would cover them all? i.e. something like:

```kotlin
val <T : StorageTask<T>.SnapshotBase> StorageTask<T>.progress: Flow<StorageProgress<T>>
    get() = callbackFlow {
        val progressListener = OnProgressListener<T> { snapshot ->
            StorageTaskScheduler.getInstance().scheduleCallback {
                trySendBlocking(StorageProgress.InProgress(snapshot))
            }
        }
        val pauseListener = OnPausedListener<T> { snapshot ->
            StorageTaskScheduler.getInstance().scheduleCallback {
                trySendBlocking(StorageProgress.Paused(snapshot))
            }
        }

        addOnProgressListener(progressListener)
        addOnPausedListener(pauseListener)

        awaitClose {
            removeOnProgressListener(progressListener)
            removeOnPausedListener(pauseListener)
        }
    }
```

Reply:
I did, but StorageTask is defined as `StorageTask<T : StorageTask.ProvideError>` and `StorageTask.ProvideError` is marked as `protected`, so I can't access it.

Reply:
Oops, nvm. I tried your implementation and it seems to work. I'll update the PR